src/components/checkbox-group/index.tsx

```diff
@@ -0,0 +1,75 @@
+import { useState, memo } from "react";
+
+import type { CheckboxGroupProps, CheckboxValue } from "./checkbox-group.types";
+
+const CheckboxGroup: React.FC<CheckboxGroupProps> = ({
+  groupTitle = "",
+  checkboxes,
+  titleCustomClasses = "",
+  containerCustomClasses = "",
+  checkboxGroupCustomClasses = "",
+  checkboxElementCustomClasses = "",
+  checkboxLabelCustomClasses = "",
+  onChange
```

Comment:
You might consider adding a checked property to it, so we can pass the initial state of each checkbox right from the data.

Reply:
Please take note that I have included the 'checked' prop for the initial state and added support for '...restProps' to accommodate additional properties for the checkbox element.